docs/proposals/compute-resource-metrics-api.md

```diff
@@ -0,0 +1,141 @@
+# Kubernetes compute resource metrics API
+
+## Goals
+
+Provide resource usage metrics on pods and nodes on the API server to be used
+by the scheduler to improve job placement, utilization, etc. and by end users
+to understand the resource utilization of their jobs.
+
+## Current state
+
+Right now, the Kubelet exports container metrics via an API endpoint. This
+information is not gathered nor served by the Kubernetes API server.
+
+## Use cases
+
+The first user will be kubectl. The resource usage data can be shown to the
+user via a periodically refreshing interface similar to `top` on Unix-like
+systems. This info could let users assign resource limits more efficiently.
+
+```
+$ kubectl top kubernetes-minion-abcd
+Pod: monitoring-heapster-abcde
```

Comment:
See:
https://github.com/GoogleCloudPlatform/kubernetes/issues/7843#issuecomment-101365459

I agree with 1 line per pod. Aggregate the data for each pod. Separate columns for cpu and memory. Just use mean.

The user can then use `kubectl describe pod` to get detailed stats per container.